crates/sui-core/src/transaction_input_checker.rs

```diff
@@ -217,7 +217,7 @@ async fn check_gas(
 /// Check all the objects used in the transaction against the database, and ensure
 /// that they are all the correct version and number.
 #[instrument(level = "trace", skip_all)]
-async fn check_objects(
+pub fn check_objects(
```

Comment:
@mystenmark, tagging you incase you have concerns of this being public

Reply:
I don't see why this would be a problem